trappy/base.py

```diff
@@ -265,6 +265,24 @@ def generate_parsed_data(self):
 
             yield data_dict
 
+    def optimize_dataframe(self):
+        """Optimize memory footprint by setting minimal data types required by
+           each column"""
+        for col in self.data_frame.columns:
+            if 'int' in str(self.data_frame[col].dtype):
+                self.data_frame[col].apply(pd.to_numeric, downcast='unsigned')
+                continue
+            if 'float' in str(self.data_frame[col].dtype):
```

Comment:
Checking the dtype kind seems more appropriate, since dtypes may be more complex and likely include "float" in their human-readable string representation, although they are not floats at all (same goes for int).
```
self.data_frame[col].kind == 'f'
```